apps/native/app/src/screens/license-scanner/license-scanner.tsx

```diff
@@ -90,127 +115,143 @@ export const LicenseScannerScreen: NavigationFunctionComponent = ({
 }) => {
   useNavigationOptions(componentId)
   useConnectivityIndicator({ componentId, rightButtons: RIGHT_BUTTONS })
-  const [hasPermission, setHasPermission] = useState<boolean>()
-  const [active, setActive] = useState(true)
-  const [invalid, setInvalid] = useState<boolean>(false)
-  const [torch, setTorch] = useState<boolean>(false)
-
-  const camera = useRef<Camera>()
-  const [layout, setLayout] = useState<LayoutRectangle>()
-  const [ratio, setRatio] = useState<string>()
-  const [padding, setPadding] = useState(0)
-
-  const invalidTimeout = useRef<ReturnType<typeof setTimeout>>()
   const intl = useIntl()
 
-  const [verifyLicenseBarcode, { data, error, loading }] =
-    useVerifyLicenseBarcodeMutation()
+  const [active, setActive] = useState(true)
+  const [invalid, setInvalid] = useState(false)
+  const [torch, setTorch] = useState(false)
 
-  useEffect(() => {
-    Camera.requestCameraPermissionsAsync().then(({ status }) => {
-      setHasPermission(status === 'granted')
-    })
-  }, [])
+  const scanningRef = useRef<ScanningStatus>(ScanningStatus.INITIAL)
```

Comment:
using scanning ref to prevent multiple rerenders when scanning since the Camera scanner callback executes very frequently